src/main/scala/nl/knaw/dans/easy/stage/dataset/Util.scala

```diff
@@ -67,6 +67,25 @@ object Util {
     } yield audience.text
   }
 
+  def readFileType(filePath: String)(implicit s: Settings): Try[Option[String]] = Try {
+    val filetype = for {
+      file <- loadBagXML("metadata/files.xml") \\ "files" \ "file"
+      if (file \ "@filepath").text == filePath
+      filetype <- file \ "type"
+    } yield filetype
+    if (filetype.size == 1) Option(filetype.head.text)
+    else None
```

Comment:
Optimisation is not an issue here, also I will refactor the code into a function that returns a Seq[Node]... The consumer can decide what to do with it then.